ostelco-ios-client/Network/Analytics.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  Analytics.swift
+//  ostelco-ios-client
+//
+//  Created by mac on 4/29/19.
+//  Copyright © 2019 mac. All rights reserved.
+//
+
+import Foundation
+import ostelco_core
+import FirebaseAnalytics
+
+class ROAnalytics {
+    static func logEvent(_ event: AnalyticsEvent) {
+        Analytics.logEvent(event.name, parameters: event.metadata)
+    }
+}
+
+enum AnalyticsEvent {
+    case LegalStuffAgreed
+    case EnteredNickname
+    case ChosenCountry(country: Country)
+    case ChosenIDMethod(idMethod: String)
+    case DownloadingESIM
+}
+
+extension AnalyticsEvent {
+    var name: String {
+        switch self {
+        case .ChosenCountry:
+            return "chosen_country"
+        case .ChosenIDMethod:
+            return "chosen_id_method"
+        default:
+            return String(describing: self).snakeCased()! // TODO: Is this safe enough to force unwrap?
```

Comment:
I would just use `...snakeCased() ?? "" - I really try to avoid force unwrapping at all in anything related to analytics because crashing in Analytics calls is a pretty major 🤦‍♀️

Reply:
I agree, that sucks for users, if app crashes because of analytics 😓 
I'll add optional and a check before sending the event, if the string is empty, maby even send a warning to crashlytics